be/src/exec/workgroup/work_group.cpp

```diff
@@ -69,14 +69,36 @@ void WorkGroup::init() {
     int64_t limit = ExecEnv::GetInstance()->query_pool_mem_tracker()->limit() * _memory_limit;
     _mem_tracker =
             std::make_shared<starrocks::MemTracker>(limit, _name, ExecEnv::GetInstance()->query_pool_mem_tracker());
-    _driver_queue = std::make_unique<starrocks::pipeline::QuerySharedDriverQueue>();
+    _driver_queue = std::make_unique<pipeline::QuerySharedDriverQueueWithoutLock>();
 }
 
-WorkGroupManager::WorkGroupManager() {}
+double WorkGroup::get_cpu_expected_use_ratio() const {
+    return static_cast<double>(_cpu_limit) / WorkGroupManager::instance()->sum_cpu_limit();
+}
+
+double WorkGroup::get_cpu_actual_use_ratio() const {
+    int64_t sum_cpu_runtime_ns = WorkGroupManager::instance()->sum_cpu_runtime_ns();
+    if (sum_cpu_runtime_ns == 0) {
+        return 0;
+    }
+    return static_cast<double>(real_runtime_ns()) / sum_cpu_runtime_ns;
+}
+
+WorkGroupManager::WorkGroupManager()
+        : _driver_dispatcher_owner_manager(std::make_unique<DispatcherOwnerManager>(
+                  config::pipeline_exec_thread_pool_thread_num > 0 ? config::pipeline_exec_thread_pool_thread_num
+                                                                   : std::thread::hardware_concurrency())),
```

Comment:
If this interface aware of the document environment ? What if BE runs in a docker, whose CPU is limited ?

Reply:
`hardware_concurrency` in Linux invokes `sysconf(_SC_NPROCESSORS_ONLN)`, which reads `/sys/devices/system/cpu/online`. If docker uses `cpuset.cpus` to limit cpu, this interface cannot be aware of this.

`hardware_concurrency` is used in lots of places in our codes. And the aim of code here is  to get the num threads of `driver_dispatcher` which is created as follows:
```C++
    auto num_threads = std::thread::hardware_concurrency();
    if (config::pipeline_exec_thread_pool_thread_num > 0) {
        num_threads = config::pipeline_exec_thread_pool_thread_num;
    }

    _driver_dispatcher = new pipeline::GlobalDriverDispatcher(num_threads, ....);
```

Therefore, maybe we could uses some plugin to modify the system file of container, like [LXCFS](https://github.com/lxc/lxcfs). Or change another way to get the number of cores in the future PR.

Reply:
Got it. Thanks.